File: 剑指Offer/05 - 替换空格/Solution.cs
/// 面试题05. 替换空格 (https://leetcode-cn.com/problems/ti-huan-kong-ge-lcof/)
/// Difficulty: Easy
/// 执行用时: 100 ms, 在所有 C# 提交中击败了 69.34% 的用户
/// 内存消耗: 24.9 MB, 在所有 C# 提交中击败了 100.00% 的用户
public class Solution
{
    public string ReplaceSpace(string s)
    {
        string updatedString = "";
        for (int i = 0; i < s.Length; i++)
        {
            updatedString += s[i] == ' ' ? "%20" : s[i].ToString();
        }

        return updatedString;
    }
}

File: 剑指Offer/05 - 替换空格/solution.ts
/**
 * 面试题05. 替换空格 (https://leetcode-cn.com/problems/ti-huan-kong-ge-lcof/)
 * Difficulty: Easy
 * 
 * 执行用时：76 ms, 在所有 typescript 提交中击败了 58.54% 的用户
 * 内存消耗：36.8 MB, 在所有 typescript 提交中击败了 94.44% 的用户
 *
 * @param {string} s
 * @returns {string}
 */
function replaceSpace(s: string): string {
  return s
    .split("")
    .map((char) => (char === " " ? "%20" : char))
    .join("");
}
